server/server.go

```diff
@@ -67,6 +71,10 @@ func NewServer(opts ...Option) (*Server, error) {
 	}
 	s.HTTPServer.Handler = mux
 
+	for _, m := range s.middleware {
```

Comment:
You should probably wrap them backwards, so that the order of chain execution matches the order they are provided by the client.